File: leetcodec++/leetcode258.cpp
//
// Created by 黄敏辉 on 2019-09-17.
//

class Solution {
public:
    int addDigits(int num) {
        while(num / 10 > 0){
            int tmp = 0;
            while (num > 0){
                tmp += num % 10;
                num /= 10;
            }
            num = tmp;
        }
        return num;
    }
};

File: leetcode258.py
class Solution(object):
    def addDigits(self, num):
        """
        :type num: int
        :rtype: int
        """
        while num //10 != 0:
            res = 0
            while num > 0:
                num, remainder = divmod(num,10)
                res += remainder
            num = res
        return num